numba/core/typeinfer.py

```diff
@@ -1479,6 +1479,14 @@ def typeof_global(self, inst, target, gvar):
             # Global array in nopython mode is constant
             typ = typ.copy(readonly=True)
 
+        if isinstance(typ, types.BaseAnonymousTuple):
+            types_w_literals = (types.Integer, types.UnicodeType)
+            types_in_tuple = (typ.dtype.types
+                              if isinstance(typ.dtype, types.UnionType)
```

Comment:
You weren't to know this but `UnionType` was I think left in by accident and IIRC we're removing it. @sklam am I remembering this correctly?

Reply:
So of what type will be the `dtype` attribute of a (heterogenous) `Tuple` if UnionType is removed?

Reply:
the `Tuple.dtype` attribute was accidental added along with `UnionType`.

Reply:
thx @sklam. I have now removed references to `Tuple.dtype` in the latest version.